templates/default/sdk_worker.sh.erb

```diff
@@ -0,0 +1,13 @@
+export HADOOP_HOME=<%= node['hops']['base_dir'] %>
+export HADOOP_PID_DIR=<%= node['hops']['logs_dir'] %>
+export HADOOP_HDFS_HOME=<%= node['hops']['base_dir'] %>
+export HADOOP_CLASSPATH=::/*:/lib/*
+export HADOOP_CONF_DIR=<%= node['hops']['conf_dir'] %>
+export CLASSPATH="<%= @hadoop_glob %>:$CLASSPATH"
+export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${JAVA_HOME}/jre/lib/amd64/server
+export PATH=$PATH:${HADOOP_HOME}/bin
+
+SUBSTR=$(echo $USER | cut -d'_' -f 1)
+ 
+bash -c "source /srv/hops/anaconda/anaconda/bin/activate /srv/hops/anaconda/anaconda/envs/$SUBSTR && ./boot $* >> $LOG_DIRS/beamsdkworker-$SUBSTR-$CONTAINER_ID.log 2>&1 "
```

Comment:
Here you didn't consider that users might be running using the base environments

Reply:
You are correct, I wrote this code when `tfx` was not part of the base environment. I need to first check in the script if the project's environment exists. Also `SUBSTR=$(echo $USER | cut -d'_' -f 1)` will fail if the project name contains the `_` character, need to fix this as well.